Source/ParleyTests/MessagesManagerTests.swift

```diff
@@ -0,0 +1,252 @@
+        setWelcomeMessage()
+        messagesManager.loadCachedData()
+        
+        let firstMessageText = "Good morning!"
+        let currentDate = Date()
+        let userMessage = createUserMessage(firstMessageText, date: currentDate)
+        
+        _ = messagesManager.add(userMessage)
+        
+        XCTAssertEqual(messagesManager.messages[0].message, MESSAGE_WELCOME_TEXT, "First message should be welcome message.")
+        XCTAssertEqual(messagesManager.messages[1].type, .date, "Second message should be date header.")
+        XCTAssertEqual(messagesManager.messages[1].message, currentDate.asDate(), "Second message should contain the date of the added message.")
+        XCTAssertEqual(messagesManager.messages[2].message, firstMessageText, "Third message should be the added message.")
+        XCTAssertEqual(messagesManager.messages[2].type, .user, "Added message should be from the user.")
+        
+        XCTAssertNil(messagesManager.lastSentMessage, "Message should not be considered as being sent.")
+    }
+    
+    func testTwoMessagesOnDifferentDaysAreSeparatedByDateIndicators() {
+        setWelcomeMessage()
+        messagesManager.loadCachedData()
+        
+        guard let yesterdayDate = getYesterdayDate() else {
+            XCTFail("Failed to create a new date from yesterday.") ; return
+        }
+        let yesterdayMessageText = "All my troubles seemed so far away"
+        let yesterdayMessage = createUserMessage(yesterdayMessageText, date: yesterdayDate)
+        _ = messagesManager.add(yesterdayMessage)
+        
+        let todayMessageText = "Good morning!"
+        let todayMessageDate = Date()
+        let todayMessage = createUserMessage(todayMessageText, date: todayMessageDate)
+        _ = messagesManager.add(todayMessage)
+        
+        XCTAssertEqual(messagesManager.messages[0].message, MESSAGE_WELCOME_TEXT, "First message should be welcome message.")
+        
+        XCTAssertEqual(messagesManager.messages[1].type, .date, "Second message should be date header.")
+        XCTAssertEqual(messagesManager.messages[1].message, yesterdayDate.asDate(), "Date indicator text should be the same as the formatted date.")
+        
+        XCTAssertEqual(messagesManager.messages[2].message, yesterdayMessageText, "Yesterday message should have the correct message text.")
+        XCTAssertEqual(messagesManager.messages[2].type, .user, "Yesterday message should be from the user.")
+        
+        XCTAssertEqual(messagesManager.messages[3].type, .date, "Fourth message should be date header.")
+        XCTAssertEqual(messagesManager.messages[3].message, todayMessageDate.asDate(), "Date indicator text should be the same as the formatted date.")
+
+        XCTAssertEqual(messagesManager.messages[4].message, todayMessageText, "Today message should have the correct message text.")
+        XCTAssertEqual(messagesManager.messages[4].type, .user, "Today message should be from the user.")
```

Comment:
Nice test, a very important one for Parley, verifying the date messages implementation.

_(Formatting the date itself belongs to its own TestSuit, so perfectly excluded here)_

_P.S.: Nice test values 🙂_